File: in_go/src/pattern_17.go
package src

import "fmt"

func Pattern_17() {
	row := 9
	column := 9
	symbol := "*"
	index := 0
	halfRow := row / 2

	for i := 0; i < row; i++ {
		for j := 0; j < column; j++ {
			if j > halfRow-index && j < halfRow+index {
				fmt.Print(" ")
			} else {
				fmt.Print(symbol)
			}
		}
		if i < halfRow {
			index++
		} else {
			index--
		}
		fmt.Println()
	}
}

File: in_rust/src/pattern_17.rs
pub fn pattern_17() {
    let row = 9;
    let column = 9;
    let symbol = "*";
    let mut index = 0;
    let half_row = row / 2;

    for i in 0..row {
        for j in 0..column {
            if j > half_row - index && j < half_row + index {
                print!(" ");
            } else {
                print!("{}", symbol)
            }
        }
        if i < half_row {
            index += 1;
        } else {
            index -= 1;
        }
        println!();
    }
}
